moz_security/modules/heka/alert/idrouter.lua

```diff
@@ -0,0 +1,183 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Heka Alert IDRouter Lookup Module
+
+idrouter provides identifier specific alert routing. It is intended to be used as a
+lookup module in the Heka alert module.
+
+The configuration of this module is used to control how alerts are routed.
+
+The subjects configuration contains any user specific alerting parameters. Each entry
+within subjects should be a key, which represents the standardized user identity string, and
+a table containing configuration parameters.
+
+At the least, each subjects entry should contain a mapfrom parameter. This is used to locate
+the correct subject entry using the lookup data subject string. Where the lookup data subject
+string matches an entry in mapfrom, this identity configuration will be used for the alert.
+
+email and IRC alerts are currently supported. For each time, three categories exist.
+
+Direct notification is used to determine how to route the alert directly to the user, and will
+be used of senduser is set to true in lookup data.
+
+Global notification is used when sendglobal is set to true in the lookup data.
+
+Error notification is used when senderror is set to true in the lookup data.
+
+A catchall setting can be set for each category for both IRC and email alerts. If the subject
+entry does not specify a given category setting, the catchall will be used for the alert.
+
+If lookup data is specified that does not contain a subject entry or does not match any known
```

Comment:
It seems a bit tedious to have to constantly manage the direct alert user list and to specify a map when X = X.  How will this work with our current lastx monitor where we send a direct alert for all users?